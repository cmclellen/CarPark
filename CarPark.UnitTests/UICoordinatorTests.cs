using ApprovalTests.Reporters;
using NUnit.Framework;
using System.IO;
using System.Text;

namespace CarPark.UnitTests
{
    [TestFixture]
    public class UICoordinatorTests
    {
        [SetUp]
        protected void Setup()
        {
            StringBuilder = new StringBuilder();
        }

        private StringBuilder StringBuilder { get; set; }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void Run_ValidInputs_CorrectResult()
        {
            // ARRANGE
            StringBuilder sb = 
                new StringBuilder()
                // Early Bird
                .AppendLine("21/03/2017 06:00")
                .AppendLine("21/03/2017 16:00")

                // Early Bird
                .AppendLine("21/03/2017 06:00")
                .AppendLine("21/03/2017 16:00")

                // Weekend Rate
                .AppendLine("19/03/2017 16:00")
                .AppendLine("19/03/2017 23:00")

                // Standard Rate (t<1h)
                .AppendLine("21/03/2017 10:00")
                .AppendLine("21/03/2017 10:30")

                // Standard Rate (1h<=t<2h)
                .AppendLine("21/03/2017 10:00")
                .AppendLine("21/03/2017 11:00")

                // Standard Rate (2h<=t<3h)
                .AppendLine("21/03/2017 10:00")
                .AppendLine("21/03/2017 12:00")

                // Standard Rate (t>=3h)
                .AppendLine("21/03/2017 10:00")
                .AppendLine("21/03/2017 13:00")

                .AppendLine("exit");
            UICoordinator sut = new UICoordinator(new StringReader(sb.ToString()), new StringWriter(StringBuilder));

            // ACT
            sut.Run();

            // ASSERT
            ApprovalTests.Approvals.Verify(StringBuilder.ToString());
        }
    }
}