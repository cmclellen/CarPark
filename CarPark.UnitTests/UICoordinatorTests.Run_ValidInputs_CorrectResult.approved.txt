All times must be in the format dd/MM/yyyy HH:mm
Type exit to end the program
Time entered: Time exited: Rate Name: Early Bird
Price: $13.00

All times must be in the format dd/MM/yyyy HH:mm
Type exit to end the program
Time entered: Time exited: Rate Name: Early Bird
Price: $13.00

All times must be in the format dd/MM/yyyy HH:mm
Type exit to end the program
Time entered: Time exited: Rate Name: Weekend Rate
Price: $10.00

All times must be in the format dd/MM/yyyy HH:mm
Type exit to end the program
Time entered: Time exited: Rate Name: Standard Rate
Price: $5.00

All times must be in the format dd/MM/yyyy HH:mm
Type exit to end the program
Time entered: Time exited: Rate Name: Standard Rate
Price: $10.00

All times must be in the format dd/MM/yyyy HH:mm
Type exit to end the program
Time entered: Time exited: Rate Name: Standard Rate
Price: $15.00

All times must be in the format dd/MM/yyyy HH:mm
Type exit to end the program
Time entered: Time exited: Rate Name: Standard Rate
Price: $20.00

All times must be in the format dd/MM/yyyy HH:mm
Type exit to end the program
Time entered: 